plugin/buycourses/src/buy_course_plugin.class.php

```diff
@@ -3311,4 +3833,444 @@ private function updateServiceSaleStatus(
             ['id = ?' => (int) $serviceSaleId]
         );
     }
+
+    /**
+     * Get the items (courses or sessions) of a coupon.
+     *
+     * @param string $couponId    The coupon ID
+     * @param int    $productType The product type
+     *
+     * @return array The item data
+     */
+    private function getItemsCoupons($couponId, $productType)
+    {
+        $couponItemTable = Database::get_main_table(self::TABLE_COUPON_ITEM);
+        if ($productType == self::PRODUCT_TYPE_COURSE) {
+            $itemTable = Database::get_main_table(TABLE_MAIN_COURSE);
+            $select = ['ci.product_id as id', 'it.title'];
+        } elseif ($productType == self::PRODUCT_TYPE_SESSION) {
+            $itemTable = Database::get_main_table(TABLE_MAIN_SESSION);
+            $select = ['ci.product_id as id', 'it.name'];
+        }
+
+        $couponFrom = "
+            $couponItemTable ci
+            INNER JOIN $itemTable it
+                on it.id = ci.product_id and ci.product_type = $productType
```

Comment:
`$productType` should be filtered in some way. `Database::escape_string($productType)` is a good way (or casting to (int) if it's an integer), or check it against a list of possible product types (if you have a function that gives you the list of all possible values).
Database::select() only filters what comes in the `'where'` array